From be380e2eecf6a0e237a93ade8c16b11640af59b2 Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Thu, 20 Nov 2025 17:06:13 -0500
Subject: [PATCH] cloud-ssh-keys-fetcher: allow system_r to transition into
 cloud_ssh_keys_fetcher_t

Allow the system_r role to transition into a cloud_ssh_keys_fetcher_t
domain. This fixes invalid context errors:

    op=security_compute_sid \
        invalid_context="system_u:system_r:cloud_ssh_keys_fetcher_t:s0-s0:c0.c1023" \
	scontext=system_u:system_r:sshd_t:s0-s0:c0.c1023 \
	tcontext=system_u:object_r:cloud_ssh_keys_fetcher_exec_t:s0 \
	tclass=process

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 policy/modules/contrib/cloud-ssh-keys-fetcher.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/contrib/cloud-ssh-keys-fetcher.te b/policy/modules/contrib/cloud-ssh-keys-fetcher.te
index c9cc9268a..211b7e7f9 100644
--- a/policy/modules/contrib/cloud-ssh-keys-fetcher.te
+++ b/policy/modules/contrib/cloud-ssh-keys-fetcher.te
@@ -15,6 +15,8 @@ type cloud_ssh_keys_fetcher_exec_t;
 
 application_domain(cloud_ssh_keys_fetcher_t, cloud_ssh_keys_fetcher_exec_t)
 
+role system_r types cloud_ssh_keys_fetcher_t;
+
 ########################################
 #
 # Local policy
-- 
2.51.2

